src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2115,6 +2115,14 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                     type = null;
                     nullability = new NullabilityInfo(CodeAnalysis.NullableAnnotation.NotAnnotated, CodeAnalysis.NullableFlowState.NotNull);
                 }
+                else if (boundExpr == highestBoundNode && boundExpr is BoundUnconvertedAddressOfOperator && boundNodeForSyntacticParent is BoundConversion { ExplicitCastInCode: true })
+                {
+                    // There was an explicit cast of an address of to a type it wasn't compatible with.
+                    type = null;
+                    convertedType = null;
+                    convertedNullability = nullability;
+                    conversion = Conversion.NoConversion;
```

Comment:
>conversion = Conversion.NoConversion; [](start = 20, length = 37)

Are we testing this aspect?

Reply:
It's verified by the utility, but I need to update both this and it to have an `Identity` conversion like `(Action)M` would have.

---
In reply to: [432045889](https://github.com/dotnet/roslyn/pull/44613#discussion_r432045889) [](ancestors = 432045889)